### PR TITLE
implement mobile testing

### PR DESCRIPTION
**Pull request by valpinkman**

### CODE REVIEW

File: mobile-test-app/ios/LLToolTests/LLToolTests.m [hunk 1]
```diff
@@ -11,7 +11,7 @@
 #import <React/RCTLog.h>
 #import <React/RCTRootView.h>
 
-#define TIMEOUT_SECONDS 3600
+#define TIMEOUT_SECONDS 900
```
Comment (gre) [inline mobile-test-app/ios/LLToolTests/LLToolTests.m]: maybe not enough

Comment (valpinkman) [reply mobile-test-app/ios/LLToolTests/LLToolTests.m]: yeah don't worry i was testing with just a few tests

**Comment by gre** [comment]
@valpinkman i don't see / understand what fails in the logs. I think we will have to console.log print the result of tests in the console too because it's not clear what fails here.

**Comment by gre** [comment]
@KhalilBellakrid do you have an idea what happens in our mobile tests? The logs are showing some scan accounts are failing

```
2020-02-04 14:40:25.891 [info][tid:com.facebook.react.JavaScript] libcore: sync(72) started. libcore:1:bitcoin:xpub6BuPWhjLqutPV8SF4RMrrn8c3t7uBZbz4CBbThpbg9GYjqRMncra9mjgSfWSK7uMDz37hhzJ8wvkbDDQQJt6VgwLoszvmPiSBtLA1bPLLSn:
2020-02-04 14:40:25.912 [info][tid:com.facebook.react.JavaScript] libcore: getOrCreateWallet 041caa3a42db5bdd125b2530c47cfbe829539b5a20a5562ec839d241c67d1862f2980d26ebffee25e4f924410c3316b397f34bd572543e72c59a7569ef9032f498_bitcoin_
2020-02-04 14:40:25.971706+0000 LLTool[10867:98913] Info: 2020-02-04T14:40:25Z+00:00 I: Building wallet instance '041caa3a42db5bdd125b2530c47cfbe829539b5a20a5562ec839d241c67d1862f2980d26ebffee25e4f924410c3316b397f34bd572543e72c59a7569ef9032f498_bitcoin_' for bitcoin with parameters: "KEYCHAIN_DERIVATION_SCHEME" -> [STRING] 44'/0'/<account>'/<node>/<address>
"BLOCKCHAIN_EXPLORER_API_ENDPOINT" -> [STRING] https://explorers.api.live.ledger.com
"BLOCKCHAIN_EXPLORER_VERSION" -> [STRING] v2
2020-02-04 14:40:26.014 [info][tid:com.facebook.react.JavaScript] libcore: getOrCreateAccount
2020-02-04 14:40:26.238 [info][tid:com.facebook.react.JavaScript] libcore: sync(72) syncCoreAccount
2020-02-04 14:40:26.248778+0000 LLTool[10867:99128] Info: 2020-02-04T14:40:26Z+00:00 I: Starting synchronization for account#0 (xpub6BuPWhjLqutPV8SF4RMrrn8c3t7uBZbz4CBbThpbg9GYjqRMncra9mjgSfWSK7uMDz37hhzJ8wvkbDDQQJt6VgwLoszvmPiSBtLA1bPLLSn) of wallet 041caa3a42db5bdd125b2530c47cfbe829539b5a20a5562ec839d241c67d1862f2980d26ebffee25e4f924410c3316b397f34bd572543e72c59a7569ef9032f498_bitcoin_ at 2020-02-04T14:40:26Z
2020-02-04 14:40:26.259980+0000 LLTool[10867:98918] Info: 2020-02-04T14:40:26Z+00:00 I: GET https://explorers.api.live.ledger.com/blockchain/v2/btc/blocks/current
2020-02-04 14:40:26.266934+0000 LLTool[10867:98918] Info: 2020-02-04T14:40:26Z+00:00 I: GET https://explorers.api.live.ledger.com/blockchain/v2/btc/syncToken
2020-02-04 14:40:26.268490+0000 LLTool[10867:98918] Info: 2020-02-04T14:40:26Z+00:00 I: GET https://explorers.api.live.ledger.com/blockchain/v2/btc/blocks/current
2020-02-04 14:40:26.283 [info][tid:com.facebook.react.JavaScript] libcore: sync(72) DONE coreAccount.synchronize
2020-02-04 14:41:26.527915+0000 LLTool[10867:99127] Task <5DF66A04-3012-42FD-A41A-886D139D9D4D>.<448> finished with error [-1001] Error Domain=NSURLErrorDomain Code=-1001 "The request timed out." UserInfo={NSUnderlyingError=0x600d2f7af210 {Error Domain=kCFErrorDomainCFNetwork Code=-1001 "(null)" UserInfo={_kCFStreamErrorCodeKey=-2102, _kCFStreamErrorDomainKey=4}}, NSErrorFailingURLStringKey=https://explorers.api.live.ledger.com/blockchain/v2/btc/blocks/current, NSErrorFailingURLKey=https://explorers.api.live.ledger.com/blockchain/v2/btc/blocks/current, _kCFStreamErrorDomainKey=4, _kCFStreamErrorCodeKey=-2102, NSLocalizedDescription=The request timed out.}
2020-02-04 14:41:26.535507+0000 LLTool[10867:99125] Task <CA2B3819-799B-43DD-8D33-20507A095920>.<449> finished with error [-1001] Error Domain=NSURLErrorDomain Code=-1001 "The request timed out." UserInfo={NSUnderlyingError=0x600d2f7aadc0 {Error Domain=kCFErrorDomainCFNetwork Code=-1001 "(null)" UserInfo={_kCFStreamErrorCodeKey=-2102, _kCFStreamErrorDomainKey=4}}, NSErrorFailingURLStringKey=https://explorers.api.live.ledger.com/blockchain/v2/btc/syncToken, NSErrorFailingURLKey=https://explorers.api.live.ledger.com/blockchain/v2/btc/syncToken, _kCFStreamErrorDomainKey=4, _kCFStreamErrorCodeKey=-2102, NSLocalizedDescription=The request timed out.}
2020-02-04 14:41:26.541834+0000 LLTool[10867:99540] Task <A8DAB531-2304-4E41-8239-0C3600A28785>.<450> finished with error [-1001] Error Domain=NSURLErrorDomain Code=-1001 "The request timed out." UserInfo={NSUnderlyingError=0x600d2f7a4030 {Error Domain=kCFErrorDomainCFNetwork Code=-1001 "(null)" UserInfo={_kCFStreamErrorCodeKey=-2102, _kCFStreamErrorDomainKey=4}}, NSErrorFailingURLStringKey=https://explorers.api.live.ledger.com/blockchain/v2/btc/blocks/current, NSErrorFailingURLKey=https://explorers.api.live.ledger.com/blockchain/v2/btc/blocks/current, _kCFStreamErrorDomainKey=4, _kCFStreamErrorCodeKey=-2102, NSLocalizedDescription=The request timed out.}
2020-02-04 14:41:26.546849+0000 LLTool[10867:99538] Task <5DF66A04-3012-42FD-A41A-886D139D9D4D>.<448> HTTP load failed, 0/0 bytes (error code: -999 [1:89])
2020-02-04 14:41:26.555741+0000 LLTool[10867:99540] Error: 2020-02-04T14:41:26Z+00:00 E: Error during during synchronization for account#0 of wallet 041caa3a42db5bdd125b2530c47cfbe829539b5a20a5562ec839d241c67d1862f2980d26ebffee25e4f924410c3316b397f34bd572543e72c59a7569ef9032f498_bitcoin_ in 60000 ms
2020-02-04 14:41:26.556048+0000 LLTool[10867:99540] Error: 2020-02-04T14:41:26Z+00:00 E: Due to EC_PRIV_KEY_INVALID_FORMAT, The request timed out. (EC_PRIV_KEY_INVALID_FORMAT)
2020-02-04 14:41:26.581 [info][tid:com.facebook.react.JavaScript] ❌ libcore > libcore bridge on account Bitcoin 1 (legacy) > sync > succeed FAILED

```

or also

```

2020-02-04 14:36:20.038399+0000 LLTool[10867:95397] Info: 2020-02-04T14:36:20Z+00:00 I: Get batch took 0:0:0.837.
2020-02-04 14:36:20.042642+0000 LLTool[10867:95397] Info: 2020-02-04T14:36:20Z+00:00 I: Transaction computation started.
2020-02-04 14:36:20.042807+0000 LLTool[10867:95397] Info: 2020-02-04T14:36:20Z+00:00 I: Got 32 txs
2020-02-04 14:36:20.349258+0000 LLTool[10867:95399] Info: 2020-02-04T14:36:20Z+00:00 I: Transaction computation took 0:0:0.310.
2020-02-04 14:36:20.350444+0000 LLTool[10867:94269] Info: 2020-02-04T14:36:20Z+00:00 I: Synchronize batch 0 took 0:0:1.152.
2020-02-04 14:36:20.357474+0000 LLTool[10867:94269] Info: 2020-02-04T14:36:20Z+00:00 I: SYNC BATCHES
2020-02-04 14:36:20.357635+0000 LLTool[10867:94269] Info: 2020-02-04T14:36:20Z+00:00 I: Synchronize batch 1 started.
2020-02-04 14:36:20.357763+0000 LLTool[10867:94269] Info: 2020-02-04T14:36:20Z+00:00 I: SYNC BATCH 1
2020-02-04 14:36:20.357877+0000 LLTool[10867:94269] Info: 2020-02-04T14:36:20Z+00:00 I: Batch derivation started.
2020-02-04 14:36:20.361187+0000 LLTool[10867:94269] Info: 2020-02-04T14:36:20Z+00:00 I: Batch derivation took 0:0:0.3.
2020-02-04 14:36:20.361341+0000 LLTool[10867:94269] Info: 2020-02-04T14:36:20Z+00:00 I: Get batch started.
2020-02-04 14:36:20.361521+0000 LLTool[10867:95399] Info: 2020-02-04T14:36:20Z+00:00 I: GET https://explorers.api.live.ledger.com/blockchain/v2/xst/addresses/SKJYZnhh2b5nYZZm3pgtVL9AqRDSYzkXUP,RzgjFu99yFCf89z39riTS3BSYy2BXtXDTb,S7jviAvUSReKJNM2BgVWrMsJmE5fXb8atz,SGvnUdpp4dvbho87LpEbdPx9wMLVLWRdMG,SFiJdpjNzLHyQ3MfUxwCrmDgwzgNmGgXTp,S27nfmUeXHzNEiA8PEnjr9MX3Etaxn22xb,SL8aEG3CM13ekiTkzmPQ3CGr4iBfLXXBPS,SHgmBzmGMykyFDWZUrypJEuTqEL87hvspn,S1HrVqiGEtKPshSMtGKM3nGR14XjbNxjA2,S77CKYxEa5H6WH4AqyQivgtStH52D9MnLS,SLgrHBoqmj1vncbF65kenvzSVLasRYaT5J,S5RGXHzJouFNJKEdw1wN6zRqPjCzKZ2QXz,RxhBBQZyEZYxVHgBzJoyPMCgQqdzLASW7M,RzgnShrFHuHm2S2pgmoGXy9yGyP97dm414,SHPwc5DZMHWzKwYHBeG2tv9BSWHTQeo1hm,S18ugEkktTh95ukEjcEhg4tNpVRdyBiYkE,S13GqHifdZekR48hkQs4efGbGKFkEzbfUv,SAdPMwF99VMpCoAsPM7G4q6LKmoQ6JzDH9,S6i7wSpacPX57ufPGYx4gdDksu7CfdbDUX,SJwTbWUWBpKdFs3FT4GU37koZKab2rCynq,S3Jhmm3jthw6cQxXfCNGqoEjSLswZS93tV,RzFfrLE8ttLtxnioowb29HbXMDbc2g9FL5,SKW4akgojdaf2bQUbimDzHjLA28riRQAfy,SASwJcbibqsysAWeAkP2CpLb72t3SHfiWC,SKQzgV7nnkJUYEbizNPn3zFkbEy3VFX6JN,S4vUB4vqSfPL4G3EWT1efwVAe3JPXRU5Sh,SAhf6Jvb3HkceEywFqR8r6qLBLg3mXVJFM,S4KcLjYdDpAe2eFhc4tFMiuRgvPC3GwRwK,SKamgDrtPyBNJXeBpCvjhUR1DFu7XtbPRU,SKNq7ggcvXiJZNWWpt5ULRPuNeAFEmaftf,Rz1RRqniL1y5WyaCxqpiD8BAe51NF2rc1j,S8pvao17Ad6TDGtWy7PTd7upuco3zv2D5p,S2KYrweQZ2LKiM5ync9Y5uknm4fNVfqA9B,S6pzeDrJzyFwadKugscsn54fTaF7BMtFik,SLVpJ8gFrE8P66vfLeEzKyS66nePpRmGcN,RxfkqhxTnnPxamxNZQcnU9S11MWyMQvbJx,SDgCzjwgSeGVPHSGFhLuxEMTZkg1FLTRcr,RzoJCvoKkfFQ6FwjuKe412fQrzhkPpTuyE,SJMTf3YY6qGvVUxPRYfbtp1HhXW9EeWcm2,RyZV3c8v8p3jbkFfpUULUjj3hoJbgAabWW/transactions
2020-02-04 14:37:20.578265+0000 LLTool[10867:97763] Task <8B3F6040-89D3-4E8E-899B-0107502FAF88>.<438> finished with error [-1001] Error Domain=NSURLErrorDomain Code=-1001 "The request timed out." UserInfo={NSUnderlyingError=0x600d1b61ea30 {Error Domain=kCFErrorDomainCFNetwork Code=-1001 "(null)" UserInfo={_kCFStreamErrorCodeKey=-2102, _kCFStreamErrorDomainKey=4}}, NSErrorFailingURLStringKey=https://explorers.api.live.ledger.com/blockchain/v2/xst/addresses/SKJYZnhh2b5nYZZm3pgtVL9AqRDSYzkXUP,RzgjFu99yFCf89z39riTS3BSYy2BXtXDTb,S7jviAvUSReKJNM2BgVWrMsJmE5fXb8atz,SGvnUdpp4dvbho87LpEbdPx9wMLVLWRdMG,SFiJdpjNzLHyQ3MfUxwCrmDgwzgNmGgXTp,S27nfmUeXHzNEiA8PEnjr9MX3Etaxn22xb,SL8aEG3CM13ekiTkzmPQ3CGr4iBfLXXBPS,SHgmBzmGMykyFDWZUrypJEuTqEL87hvspn,S1HrVqiGEtKPshSMtGKM3nGR14XjbNxjA2,S77CKYxEa5H6WH4AqyQivgtStH52D9MnLS,SLgrHBoqmj1vncbF65kenvzSVLasRYaT5J,S5RGXHzJouFNJKEdw1wN6zRqPjCzKZ2QXz,RxhBBQZyEZYxVHgBzJoyPMCgQqdzLASW7M,RzgnShrFHuHm2S2pgmoGXy9yGyP97dm414,SHPwc5DZMHWzKwYHBeG2tv9BSWHTQeo1hm,S18ugEkktTh95ukEjcEhg4tNpVRdyBiYkE,S13GqHifdZekR48hkQs4efGbGKFkEzbfUv,SAdPMwF99VMpCoAsPM7G4q6LKmoQ6JzDH9,S6i7wSpacPX57ufPGYx4gdDksu7CfdbDUX,SJwTbWUWBpKdFs3FT4GU37koZKab2rCynq,S3Jhmm3jthw6cQxXfCNGqoEjSLswZS93tV,RzFfrLE8ttLtxnioowb29HbXMDbc2g9FL5,SKW4akgojdaf2bQUbimDzHjLA28riRQAfy,SASwJcbibqsysAWeAkP2CpLb72t3SHfiWC,SKQzgV7nnkJUYEbizNPn3zFkbEy3VFX6JN,S4vUB4vqSfPL4G3EWT1efwVAe3JPXRU5Sh,SAhf6Jvb3HkceEywFqR8r6qLBLg3mXVJFM,S4KcLjYdDpAe2eFhc4tFMiuRgvPC3GwRwK,SKamgDrtPyBNJXeBpCvjhUR1DFu7XtbPRU,SKNq7ggcvXiJZNWWpt5ULRPuNeAFEmaftf,Rz1RRqniL1y5WyaCxqpiD8BAe51NF2rc1j,S8pvao17Ad6TDGtWy7PTd7upuco3zv2D5p,S2KYrweQZ2LKiM5ync9Y5uknm4fNVfqA9B,S6pzeDrJzyFwadKugscsn54fTaF7BMtFik,SLVpJ8gFrE8P66vfLeEzKyS66nePpRmGcN,RxfkqhxTnnPxamxNZQcnU9S11MWyMQvbJx,SDgCzjwgSeGVPHSGFhLuxEMTZkg1FLTRcr,RzoJCvoKkfFQ6FwjuKe412fQrzhkPpTuyE,SJMTf3YY6qGvVUxPRYfbtp1HhXW9EeWcm2,RyZV3c8v8p3jbkFfpUULUjj3hoJbgAabWW/transactions, NSErrorFailingURLKey=https://explorers.api.live.ledger.com/blockchain/v2/xst/addresses/SKJYZnhh2b5nYZZm3pgtVL9AqRDSYzkXUP,RzgjFu99yFCf89z39riTS3BSYy2BXtXDTb,S7jviAvUSReKJNM2BgVWrMsJmE5fXb8atz,SGvnUdpp4dvbho87LpEbdPx9wMLVLWRdMG,SFiJdpjNzLHyQ3MfUxwCrmDgwzgNmGgXTp,S27nfmUeXHzNEiA8PEnjr9MX3Etaxn22xb,SL8aEG3CM13ekiTkzmPQ3CGr4iBfLXXBPS,SHgmBzmGMykyFDWZUrypJEuTqEL87hvspn,S1HrVqiGEtKPshSMtGKM3nGR14XjbNxjA2,S77CKYxEa5H6WH4AqyQivgtStH52D9MnLS,SLgrHBoqmj1vncbF65kenvzSVLasRYaT5J,S5RGXHzJouFNJKEdw1wN6zRqPjCzKZ2QXz,RxhBBQZyEZYxVHgBzJoyPMCgQqdzLASW7M,RzgnShrFHuHm2S2pgmoGXy9yGyP97dm414,SHPwc5DZMHWzKwYHBeG2tv9BSWHTQeo1hm,S18ugEkktTh95ukEjcEhg4tNpVRdyBiYkE,S13GqHifdZekR48hkQs4efGbGKFkEzbfUv,SAdPMwF99VMpCoAsPM7G4q6LKmoQ6JzDH9,S6i7wSpacPX57ufPGYx4gdDksu7CfdbDUX,SJwTbWUWBpKdFs3FT4GU37koZKab2rCynq,S3Jhmm3jthw6cQxXfCNGqoEjSLswZS93tV,RzFfrLE8ttLtxnioowb29HbXMDbc2g9FL5,SKW4akgojdaf2bQUbimDzHjLA28riRQAfy,SASwJcbibqsysAWeAkP2CpLb72t3SHfiWC,SKQzgV7nnkJUYEbizNPn3zFkbEy3VFX6JN,S4vUB4vqSfPL4G3EWT1efwVAe3JPXRU5Sh,SAhf6Jvb3HkceEywFqR8r6qLBLg3mXVJFM,S4KcLjYdDpAe2eFhc4tFMiuRgvPC3GwRwK,SKamgDrtPyBNJXeBpCvjhUR1DFu7XtbPRU,SKNq7ggcvXiJZNWWpt5ULRPuNeAFEmaftf,Rz1RRqniL1y5WyaCxqpiD8BAe51NF2rc1j,S8pvao17Ad6TDGtWy7PTd7upuco3zv2D5p,S2KYrweQZ2LKiM5ync9Y5uknm4fNVfqA9B,S6pzeDrJzyFwadKugscsn54fTaF7BMtFik,SLVpJ8gFrE8P66vfLeEzKyS66nePpRmGcN,RxfkqhxTnnPxamxNZQcnU9S11MWyMQvbJx,SDgCzjwgSeGVPHSGFhLuxEMTZkg1FLTRcr,RzoJCvoKkfFQ6FwjuKe412fQrzhkPpTuyE,SJMTf3YY6qGvVUxPRYfbtp1HhXW9EeWcm2,RyZV3c8v8p3jbkFfpUULUjj3hoJbgAabWW/transactions, _kCFStreamErrorDomainKey=4, _kCFStreamErrorCodeKey=-2102, NSLocalizedDescription=The request timed out.}
2020-02-04 14:37:20.590468+0000 LLTool[10867:97763] Info: 2020-02-04T14:37:20Z+00:00 I: Recovering from failing synchronization : The request timed out. (EC_PRIV_KEY_INVALID_FORMAT)
2020-02-04 14:37:20.590903+0000 LLTool[10867:97781] Info: 2020-02-04T14:37:20Z+00:00 I: Recovering from failing synchronization : The request timed out. (EC_PRIV_KEY_INVALID_FORMAT)
2020-02-04 14:37:20.595095+0000 LLTool[10867:97781] Error: 2020-02-04T14:37:20Z+00:00 E: Error during during synchronization for account#2 of wallet 0488aadaf45765f921843995645769b6a75aa0f03a8a91aa935968443a107aab543a213f197ee9c9fe3f7aa6679fa3004c0782b1740dc706d77bb4da28bd5433b8_stealthcoin_ in 62000 ms
2020-02-04 14:37:20.595721+0000 LLTool[10867:97781] Error: 2020-02-04T14:37:20Z+00:00 E: Due to EC_PRIV_KEY_INVALID_FORMAT, The request timed out. (EC_PRIV_KEY_INVALID_FORMAT)
2020-02-04 14:37:20.723 [info][tid:com.facebook.react.JavaScript] ❌ libcore > stealthcoin currency bridge > scanAccounts > stealthcoin seed 1 FAILED
```


Did we just had bad luck on HTTP? why would it work fine on Node implementation tho which runs the exact same tests and almost never fails.

maybe the http implementation on ios is not that robust?

**Comment by gre** [comment]
@valpinkman can you provide the "debian" test part in another PR? because it seems unrelated to mobile.

**Comment by valpinkman** [comment]
will do that

**Comment by valpinkman** [comment]
> @valpinkman can you provide the "debian" test part in another PR? because it seems unrelated to mobile.

https://github.com/LedgerHQ/ledger-live-common/pull/496

**Pull request closed without merging**